test/els_progress_SUITE.erl

```diff
@@ -0,0 +1,136 @@
+%% Testcases
+%%==============================================================================
+-export([ sample_job/1
+        , failing_job/1
+        ]).
+
+%%==============================================================================
+%% Includes
+%%==============================================================================
+-include_lib("common_test/include/ct.hrl").
+-include_lib("stdlib/include/assert.hrl").
+
+-include("erlang_ls.hrl").
+
+%%==============================================================================
+%% Types
+%%==============================================================================
+-type config() :: [{atom(), any()}].
+
+%%==============================================================================
+%% CT Callbacks
+%%==============================================================================
+-spec suite() -> [tuple()].
+suite() ->
+  [{timetrap, {seconds, 30}}].
+
+-spec all() -> [{group, atom()}].
+all() ->
+  [{group, tcp}, {group, stdio}].
+
+-spec groups() -> [atom()].
+groups() ->
+  els_test_utils:groups(?MODULE).
+
+-spec init_per_suite(config()) -> config().
+init_per_suite(Config) ->
+  els_test_utils:init_per_suite(Config).
+
+-spec end_per_suite(config()) -> ok.
+end_per_suite(Config) ->
+  els_test_utils:end_per_suite(Config).
+
+-spec init_per_testcase(atom(), config()) -> config().
+init_per_testcase(sample_job = TestCase, Config) ->
+  Task = fun(_) -> ok end,
+  setup_mocks(Task),
+  [{task, Task} | els_test_utils:init_per_testcase(TestCase, Config)];
+init_per_testcase(failing_job = TestCase, Config) ->
+  Task = fun(_) -> exit(fail) end,
+  setup_mocks(Task),
+  [{task, Task} | els_test_utils:init_per_testcase(TestCase, Config)].
+
+-spec end_per_testcase(atom(), config()) -> ok.
+end_per_testcase(TestCase, Config) ->
+  els_test_utils:end_per_testcase(TestCase, Config),
+  teardown_mocks().
+
+%%==============================================================================
+%% Testcases
+%%==============================================================================
+
+-spec sample_job(config()) -> ok.
+sample_job(_Config) ->
+  {ok, Pid} = new_background_job(),
+  wait_for_completion(Pid),
+  ?assertEqual(length(entries()), meck:num_calls(sample_job, task, '_')),
+  ?assertEqual(1, meck:num_calls(sample_job, on_complete, '_')),
+  ?assertEqual(0, meck:num_calls(sample_job, on_error, '_')),
+  ok.
+
+-spec failing_job(config()) -> ok.
+failing_job(_Config) ->
+  {ok, Pid} = new_background_job(),
+  wait_for_completion(Pid),
+  ?assertEqual(1, meck:num_calls(sample_job, task, '_')),
+  ?assertEqual(0, meck:num_calls(sample_job, on_complete, '_')),
+  ?assertEqual(1, meck:num_calls(sample_job, on_error, '_')),
+  ok.
+
+%%==============================================================================
+%% Internal Functions
+%%==============================================================================
+-spec wait_for_completion(pid()) -> ok.
+wait_for_completion(Pid) ->
```

Comment:
Could be useful to move to `els_test_utils`.

Reply:
Probably. Let's do it once we have a second use case, though.

Reply:
I doubt I will remember this function exists if/when I need it 😞.